src/server/service/kv.rs

```diff
@@ -300,22 +309,24 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
             return;
         }
         let begin_instant = Instant::now_coarse();
-        let future = future_cop(&self.cop, Some(ctx.peer()), req)
-            .and_then(|resp| sink.success(resp).map_err(Error::from))
-            .map(move |_| {
-                GRPC_MSG_HISTOGRAM_STATIC
-                    .coprocessor
-                    .observe(duration_to_sec(begin_instant.elapsed()))
-            })
-            .map_err(move |e| {
-                debug!("kv rpc failed";
-                    "request" => "coprocessor",
-                    "err" => ?e
-                );
-                GRPC_MSG_FAIL_COUNTER.coprocessor.inc();
-            });
+        let future = future_cop(&self.cop, Some(ctx.peer()), req);
+        let task = async move {
+            let resp = future.await?;
+            sink.success(resp).compat().map_err(Error::from).await?;
+            GRPC_MSG_HISTOGRAM_STATIC
+                .coprocessor
+                .observe(duration_to_sec(begin_instant.elapsed()));
+            ServerResult::Ok(())
+        }
+        .map_err(|e| {
+            debug!("kv rpc failed";
+                "request" => "coprocessor",
+                "err" => ?e
+            );
+            GRPC_MSG_FAIL_COUNTER.coprocessor.inc();
+        });
 
-        ctx.spawn(future);
+        ctx.spawn(Compat::new(task.boxed()));
```

Comment:
Otherwise compiler will complain: 
```
   --> src/server/service/kv.rs:329:13
    |
329 |         ctx.spawn(Compat::new(task));
    |             ^^^^^ within `impl failure::core::future::future::Future`, the trait `std::marker::Unpin` is not implemented for `std::future::from_generator::GenFuture<[static generator@src/server/service/kv.rs:313:31: 320:10 future:_, sink:_, begin_instant:_ _]>`
```

Reply:
Is there any way to make task implement `Unpin` automatically?

Reply:
async blocks are always `!Unpin`.

Reply:
`Unpin` is the requirement of compat (I don't know why, however). 
A typical future executor should accept an `!Unpin` future as the executor will box it. So after we don't need that compat, we should get rid of the extra box too.